cmd/agent/app/reporter/metrics_test.go

```diff
@@ -26,24 +26,12 @@ import (
 	"github.com/jaegertracing/jaeger/thrift-gen/zipkincore"
 )
 
-type noopReporter struct {
-	err error
-}
-
-func (r *noopReporter) EmitZipkinBatch(spans []*zipkincore.Span) error {
-	return r.err
-}
-
-func (r *noopReporter) EmitBatch(batch *jaeger.Batch) error {
-	return r.err
-}
-
 func TestMetricsReporter(t *testing.T) {
 	tests := []struct {
 		expectedCounters []metricstest.ExpectedMetric
 		expectedGauges   []metricstest.ExpectedMetric
 		action           func(reporter Reporter)
-		rep              *noopReporter
+		rep              *alwaysFailReporter
```

Comment:
Perhaps it needs a better name? :-)

Reply:
why? it's an existing mock struct that was re-implemented twice, I just consolidated them.

Reply:
`alwaysFailReporter` is misleading: people have to look at the implementation to realize that it won't actually cause any failures, that that the `TestMetricsReporter` isn't (only) testing reporters that fail...

Reply:
renamed to mockReporter